lms/djangoapps/discussion_api/api.py

```diff
@@ -63,3 +65,58 @@ def get_module_sort_key(module):
         "courseware_topics": courseware_topics,
         "non_courseware_topics": non_courseware_topics,
     }
+
+
+def _cc_thread_to_api_thread(thread):
+    """
+    Convert a thread data dict from the comment_client format (which is a direct
+    representation of the format returned by the comments service) to the format
+    used in this API
+    """
+    ret = {
+        key: thread[key]
+        for key in [
+            "id",
+            "course_id",
+            "created_at",
+            "updated_at",
+            "type",
+            "title",
+            "pinned",
+            "closed",
+        ]
+    }
+    ret.update({
```

Comment:
Generally, the Discussion API is intended for external consumption, so it is designed to make sense in itself and to be as free as possible from any weirdness imposed by the existing implementation details.

More specifically:
"commentable" is not a term that makes sense to anybody except those who have worked on the forums within edX.
"raw_body" is distinguished from "rendered_body", which will come in a later commit.
"comments_count" and "unread_comments_count" are just weird phrasing.